### PR TITLE
Fixes the problem with folders and files in the workdir permissions

### LINKED ISSUE #2355: Configurable permissions
At the moment, toil sets pretty stringent permissions in the jobstore, e.g. a pipeline can't be restarted by a user different from the original user. In shared HPC environments is plausible to have multiple engineers running the operations. Would be nice if some how one could set default permissions used for files created within the jobstore. Even better if one could tell toil to honor the current `umask` configuration.

┆Issue is synchronized with this [Jira Story](https://ucsc-cgl.atlassian.net/browse/TOIL-273)
┆Issue Number: TOIL-273

### PR DESCRIPTION
**Pull request by mberacochea**
Instead of using python tempfile to create files and dirs in the
work dir use regular files with uniq names. This allows those files
and folders to honor the system permissions and umask.

This will fix #2355

### CODE REVIEW

**Review by DailyDreaming — CHANGES_REQUESTED** [review]
Other than the comment below, this seems alright.  Thanks for submitting the fix!

**Comment by adamnovak** [comment]
I snuck in and patched this. Let's see if it works.

**Comment by adamnovak** [comment]
Adding in UUIDs might make all our filenames longer in the job store. We'll have to see if that turns out to be a debuggability problem.

File: src/toil/jobStores/fileJobStore.py [hunk 1]
```diff
@@ -565,11 +566,11 @@ def readSharedFileStream(self, sharedFileName):
 
     def writeStatsAndLogging(self, statsAndLoggingString):
         # Temporary files are placed in the stats directory tree
-        fd, tempStatsFile = tempfile.mkstemp(prefix="stats", suffix=".new", dir=self._getArbitraryStatsDir())
+        tempStatsFileName = "stats" + str(uuid.uuid4().hex) + ".new"
+        tempStatsFile = os.path.join(self._getArbitraryStatsDir, tempStatsFileName)
```
[the remark this answers is not in the document]
Comment (mberacochea) [reply src/toil/jobStores/fileJobStore.py]: Ugh, silly mistake. Thanks for catching that one.

**Comment by mberacochea** [comment]
@adamnovak OK, that make sense. I can re-write this and use the first 8 chars from the UUID but checking existing folders to avoid collisions.

**Comment by adamnovak** [comment]
I don't think truncating to 8 characters is a good idea. We should leave the UUIDs as full UUIDs.

The `mktemp`-type functions do some fancy work to ensure that you actually get a filename that doesn't already exist, to prevent collisions. If we're only using 8 **completely random** hex characters, we only have 32 bits of entropy (vs. 128 bits or something with a full UUID), and [collisions start to become possible with physically-achievable numbers of files](https://stackoverflow.com/a/14210379/402891).

**Comment by mberacochea** [comment]
> I don't think truncating to 8 characters is a good idea. We should leave the UUIDs as full UUIDs.
> 
> The `mktemp`-type functions do some fancy work to ensure that you actually get a filename that doesn't already exist, to prevent collisions. If we're only using 8 **completely random** hex characters, we only have 32 bits of entropy (vs. 128 bits or something with a full UUID), and [collisions start to become possible with physically-achievable numbers of files](https://stackoverflow.com/a/14210379/402891).

OK. I agree, I'll revert that commit.

**Comment by mberacochea** [comment]
Done. :)

> > I don't think truncating to 8 characters is a good idea. We should leave the UUIDs as full UUIDs.
> > The `mktemp`-type functions do some fancy work to ensure that you actually get a filename that doesn't already exist, to prevent collisions. If we're only using 8 **completely random** hex characters, we only have 32 bits of entropy (vs. 128 bits or something with a full UUID), and [collisions start to become possible with physically-achievable numbers of files](https://stackoverflow.com/a/14210379/402891).
> 
> OK. I agree, I'll revert that commit.

Done :)

**Review by adamnovak — APPROVED** [review]
Looks like this should work.

**Comment by adamnovak** [comment]
@DailyDreaming I'm going to declare your review addressed and merge this.

**Pull request merged**